src/components/InfinteScroller.tsx

```diff
@@ -0,0 +1,40 @@
+import useInfiniteScroll from "@/hooks/useInfiniteScroll";
+import { InfiniteScrollerProps } from "@/interfaces/interfaces";
+import React from "react";
+import InfiniteScroll from "react-infinite-scroll-component";
+import EventList from "./EventList";
+import Loading from "../../public/icons/Loading.svg";
+
+function InfiniteScroller({ events, edit, limit }: InfiniteScrollerProps) {
+  const { loadedEvents, hasMore, fetchMoreData, loading } =
+    useInfiniteScroll(events);
+
+  return (
+    <>
+      {loading ? (
+        <div className="flex justify-center items-center h-screen">
+          <h2 className="text-center">Loading initial events...</h2>
+          <Loading className="inline mr-2 w-4 h-4 text-gray-200 animate-spin dark:text-gray-600" />
+        </div>
+      ) : (
+        <InfiniteScroll
+          dataLength={loadedEvents.length}
+          next={() => fetchMoreData(limit)}
+          hasMore={hasMore}
+          loader={<h4 className="text-center">Loading more events...</h4>}
```

Comment:
Can you add top and bottom margins here so that the loader is more visible below when fetching more items?